pxr/usdImaging/usdImagingGL/CMakeLists.txt

```diff
@@ -142,3 +146,4432 @@ pxr_build_test(testUsdImagingGLSurfaceShader
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLInvalidUniversalSurface
+    DEST testUsdImagingGLInvalidUniversalSurface
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLFallbackPrimTypes
+    DEST testUsdImagingGLFallbackPrimTypes
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLPrimvarProcessing
+    DEST testUsdImagingGLPrimvarProcessing
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLComposite
+    DEST testUsdImagingGLComposite
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLPresentDisabled
+    DEST testUsdImagingGLPresentDisabled
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLFraming
+    DEST testUsdImagingGLFraming
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLHWFaceCulling
+    DEST testUsdImagingGLHWFaceCulling
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLMipmap
+    DEST testUsdImagingGLMipmap
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLAovVisualization
+    DEST testUsdImagingGLAovVisualization
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLPrimvarRefinement
+    DEST testUsdImagingGLPrimvarRefinement
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLDisplacement
+    DEST testUsdImagingGLDisplacement
+)
+
+pxr_install_test_dir(
+    SRC testenv/testUsdImagingGLGeomSubsets
+    DEST testUsdImagingGLGeomSubsets
+)
+
+#
+# Register all the tests
+#
+#pxr_register_test(testUsdImagingGLImageDiff
+#    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -offscreen -stage test.usda"
+#    IMAGE_DIFF_COMPARE
+#        different_by_one_percent.png
+#    EXPECTED_RETURN_CODE 1
+#    TESTENV testUsdImagingGLImageDiff
+#)
+#
+#pxr_register_test(testUsdImagingGLImageDiff_FailPercent
+#    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -offscreen -stage test.usda"
+#    IMAGE_DIFF_COMPARE
+#        different_by_one_percent.png
+#    FAIL_PERCENT 2
+#    EXPECTED_RETURN_CODE 0
+#    TESTENV testUsdImagingGLImageDiff
+#)
+#
+#pxr_register_test(testUsdImagingGLImageDiff_Fail
+#    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -offscreen -stage test.usda"
+#    IMAGE_DIFF_COMPARE
+#        different_by_one_percent.png
+#    FAIL 0.75
+#    EXPECTED_RETURN_CODE 0
+#    TESTENV testUsdImagingGLImageDiff
+#)
+#
+#pxr_register_test(testUsdImagingGLImageDiff_Scale
+#    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -offscreen -stage test.usda"
+#    IMAGE_DIFF_COMPARE
+#        different_by_scale.png
+#    EXPECTED_RETURN_CODE 1
+#    TESTENV testUsdImagingGLImageDiff
+#)
+
+pxr_register_test(testUsdImagingGLBasicDrawing
+    COMMAND "${CMAKE_INSTALL_PREFIX}/tests/testUsdImagingGLBasicDrawing -offscreen -stage basicDrawing/basicDrawing.usda -write testUsdImagingGLBasicDrawing.png"
+    IMAGE_DIFF_COMPARE
+        testUsdImagingGLBasicDrawing.png
+    FAIL 1
+    FAIL_PERCENT 1
+    PERCEPTUAL
```

Comment:
This is definitely a point worth discussing. At the moment, `idiff` is being used to perform the actual image diff. However,  the `perceptual` option on `idiff` is run in addition to the normal `fail` / `failpercent` tests. But as far as I could tell, the `perceptualdiff` from testSpecs.xml only tested for a perceptual difference. So to stick with that same behavior I've set `FAIL` to 1 to ignore that part of the test and only perform the perceptual test. I think the other option here would be put this logic inside of `testWrapper.py` where `idiff` is wrapped.

Reply:
Great point! I was debating about this but ultimately I think setting `FAIL` to 1 in the tests is better. This way someone can see exactly what arguments we're passing to `idiff` rather than have to dig into `testWrapper.py` too. Having the logic in `testWrapper.py` would help if we were trying to abstract away the fact that we're using `idiff`, but that doesn't seem super important right now.